src/http-client.ts

```diff
@@ -27,18 +27,21 @@ export class HttpClient {
     const retry = retryPromise<T>(
       this.requestApi.bind(this, path, opts),
       (error: any) => error && HttpClient.retryStatusCodes.has(error.status),
-      retryInterval
+      retryInterval,
+      opts.logger
     );
 
     this._httpRetryingRequests.set(retry._id, retry);
 
     /* tslint:disable:no-floating-promises */
-    retry.promise.finally(() => this.cancelRetryRequest(retry._id));
+    retry.promise
```

Comment:
Is this to add one more retry?

Reply:
Ahh ok I see. I would suggest changing this just slightly then because if the `.then()` fails, it's going to try again in the catch which is why I was confused. To do *exactly* what you're saying, I think you'd want to do something like:
`.then(() => this.cancelRetryRequest(retry._id), () => this.cancelRetryRequest(retry._id))` without a `.catch()`